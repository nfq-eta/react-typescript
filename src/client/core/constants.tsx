export const ACTION = {
    CHECKBOX: {
        ADD: 'ac@checkbox.add',
    },
};
