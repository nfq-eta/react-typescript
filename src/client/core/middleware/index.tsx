import logger from 'redux-logger';
export {
    logger,
};
