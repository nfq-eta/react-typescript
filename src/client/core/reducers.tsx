import { combineReducers } from 'redux';
import checkBoxReducers from '../modules/checkbox/reducers';
import { IItem } from '../components/checkbox/CheckBox';

export interface IRootState {
    checkBoxItems: IItem[];
}

const rootReducer = combineReducers({
    checkBoxItems: checkBoxReducers,
});

export default rootReducer;
