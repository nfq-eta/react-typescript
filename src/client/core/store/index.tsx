import { store as createStore } from 'react-easy-state';
import { IItem } from '../../components/checkbox/CheckBoxComponent';

export type rootStore = {
  foo: string;
  checkBoxItems: IItem[];
};

export const rootStore: rootStore = {
  foo: 'buzz',
  checkBoxItems: [],
};

export const store = createStore({
  ...rootStore,
  // tslint:disable-next-line
  ...typeof window !== 'undefined' && window && window.__PRELOADED_STATE__,
  addAction(item: IItem) {
    store.checkBoxItems.push(item);
  },
});
