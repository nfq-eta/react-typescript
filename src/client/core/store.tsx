import { createStore, applyMiddleware, Store } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { logger } from './middleware';

import rootReducer, { IRootState } from './reducers';
import { IItem } from '../components/CheckBox';

export function configureStore(initialState?: IRootState) {
    let middleware = applyMiddleware(logger);

    if (process.env.NODE_ENV === 'development') {
        middleware = composeWithDevTools(middleware);
    }

    const store = createStore(rootReducer, initialState, middleware) as Store<IItem[]>;

    if (module.hot) {
        module.hot.accept('./reducers', () => {
            const nextReducer = require('./reducers');
            store.replaceReducer(nextReducer);
        });
    }

    return store;
}
