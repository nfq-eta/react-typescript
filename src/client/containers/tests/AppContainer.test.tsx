import * as React from 'react';
import { shallow } from 'enzyme';
import AppConnected, { App } from '../AppContainer';

jest.mock('../../core/store');
import { store } from '../../core/store';

describe('AppContainer.tsx', () => {
  const props = {
    items: [{ id: 'demo1', label: 'Demo', value: 'demo' }, { id: 'demo2', label: 'Demo2', value: 'demo2' }],
  };

  it('renders without crashing', () => {
    shallow(<AppConnected {...props} />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<AppConnected {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Buttons should be visible', () => {
    const wrapper = shallow(<AppConnected {...props} />);
    expect(wrapper.find('button').length).toBe(1);
  });

  it('handleAdd()', () => {
    const wrapper = shallow(<AppConnected />).instance() as App;

    wrapper.handleAdd();
    // expect(store.addAction).toBeCalled();

    expect(store.addAction).toBeCalledWith({
      id: expect.any(String),
      label: 'Demo',
      value: 'demo',
    });
  });
});
