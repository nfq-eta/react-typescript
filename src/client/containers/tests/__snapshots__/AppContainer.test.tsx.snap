// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer.tsx matches snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Add more
  </button>
</div>
`;
